src/lib/mongoBackend/MongoCommonUpdate.cpp

```diff
@@ -1509,7 +1513,27 @@ static bool addTriggeredSubscriptions_noCache
         std::string geometry = expr.hasField(CSUB_EXPR_GEOM)   ? getStringField(expr, CSUB_EXPR_GEOM)   : "";
         std::string coords   = expr.hasField(CSUB_EXPR_COORDS) ? getStringField(expr, CSUB_EXPR_COORDS) : "";
 
-        trigs->fillExpression(q, georel, geometry, coords);
+        trigs->fillExpression(georel, geometry, coords);
+
+        // Parsing q
+        if (q != "")
+        {
+          StringFilter* stringFilterP = new StringFilter();
+
+          if (stringFilterP->parse(q.c_str(), &err) == false)
+          {
+            delete stringFilterP;
+          
+            LM_E(("Runtime Error (%s)", err.c_str()));
+            TIME_STAT_MONGO_READ_WAIT_STOP();
+            releaseMongoConnection(connection);
+            return false;
+          }
+          else
+          {
+            trigs->stringFilterSet(stringFilterP);
```

Comment:
I don't fully understand this comment.

if (q != "") => if we have a string filter that is not empty

After that, the parse of the string filter fails, and as this 'q' came from the database, it is not a 'Bad Input' but an error in the broker accepting it in the first place.
This will hopefully never ever occur.